### PR TITLE
move taper outside trim window

### PR DESCRIPTION
**Pull request by awech**

### CODE REVIEW

File: new_array_processing.py [hunk 1]
```diff
@@ -84,24 +83,26 @@ def get_volcano_backazimuth(latlist, lonlist, volcanoes):
...
+                                          STARTTIME - 2 * config.taper_val,
+                                          ENDTIME + 2 * config.taper_val,
+                                          attach_response=True)
             except header.FDSNNoDataException:
                 print(f"No data retrieved for {STA}")
                 continue
 
             st.merge(fill_value='latest')
-            st.trim(STARTTIME, ENDTIME, pad='true', fill_value=0)
+            st.trim(STARTTIME - 2 * config.taper_val, ENDTIME + 2 * config.taper_val, pad='true', fill_value=0)
```
Comment (ibrewster) [inline new_array_processing.py:95]: So you trim it to *less* than the requested start/end times? Not just *to* the requested start/end?

File: new_array_processing.py [hunk 2]
```diff
@@ -84,24 +83,26 @@ def get_volcano_backazimuth(latlist, lonlist, volcanoes):
...
             st.sort()
             print(st)
 
             print('Removing sensitivity...')
             st.remove_sensitivity()
 
             stf = st.copy()
+            stf.taper(max_percentage=None, max_length=config.taper_val)
```
Comment (ibrewster) [inline new_array_processing.py:104]: So if we are just going to trim the taper off, why do it at all? Why not just skip this step?

**Review by ibrewster — COMMENTED** [review]
Couple of comments, mostly for my edification.

File: new_array_processing.py (hunk 2 above)
Comment (awech) [reply new_array_processing.py:104]: A taper reduces artificial edge effects caused by filtering, but we don't want to plot a tapered trace. So basically the point here is to download data that extends before our desired STARTTIME and after our desired ENDTIME. Then we apply the taper outside those bounds. After we filter, we trim the trace to STARTTIME/ENDTIME, which removes the taper part.

Comment (ibrewster) [reply new_array_processing.py:104]: Ah, artificial edge effects. Gotcha, that makes sense.

File: new_array_processing.py (hunk 1 above)
Comment (ibrewster) [reply new_array_processing.py:95]: Oh, wait. Got my signs backwards. That makes sense now.

**Pull request merged**